src/components/tables/Table.tsx

```diff
@@ -0,0 +1,337 @@
+};
+
+export interface ITableProps extends ITableOwnProps, ITableStateProps, Partial<ITableDispatchProps> { }
+
+export class Table extends React.Component<ITableProps, any> {
+  componentDidMount() {
+    if (this.props.onDidMount) {
+      this.props.onDidMount();
+    }
+  }
+
+  componentWillReceiveProps(nextProps: ITableProps) {
+    const { tableState } = this.props;
+
+    if (this.hasTableStateChanged(tableState, nextProps.tableState)) {
+      const shouldResetPage =
+        tableState.page === nextProps.tableState.page
+        || tableState.perPage !== nextProps.tableState.perPage;
+
+      this.props.onModifyData(shouldResetPage);
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.props.onUnmount) {
+      this.props.onUnmount();
+    }
+  }
+
+  hasTableStateChanged(currentTableState: ITableState, nextTableState: ITableState): boolean {
+    return !!currentTableState && (
+      currentTableState.filter !== nextTableState.filter
+      || currentTableState.perPage !== nextTableState.perPage
+      || currentTableState.page !== nextTableState.page
+      || currentTableState.sortState.attribute !== nextTableState.sortState.attribute
+      || currentTableState.sortState.order !== nextTableState.sortState.order
+      || (_.isEmpty(currentTableState.predicates) && !_.isEmpty(nextTableState.predicates))
+      || _.some(
+        currentTableState.predicates,
+        (attributeValue: any, attributeName: string) => attributeValue !== nextTableState.predicates[attributeName],
+      )
+    );
+  }
+
+  buildLoadingTable(): JSX.Element {
+    return <LoadingTableConnected
+      id={getTableChildComponentId(this.props.id, TableChildComponent.LOADING_TABLE)}
+      hide={!this.props.tableState.isLoading}
+      columnsPerRow={this.props.headingAttributes.length + 1}
+      numberOfRows={
+        this.props.tableState.data && this.props.tableState.data.displayedIds.length
+        || this.props.tableState.perPage
+      } />;
+  }
+
+  buildLoadingNavigation(): JSX.Element {
+    return <LoadingConnected
+      id={getTableChildComponentId(this.props.id, TableChildComponent.LOADING_NAVIGATION)}
+      hide={!this.props.tableState.isLoading} />;
+  }
+
+  buildActionBar(): JSX.Element {
+    const { actionBar, filter, predicates } = this.props;
+
+    const filterBoxConnected: JSX.Element = actionBar && filter
+      ? <FilterBoxConnected
+        {...filter}
+        id={getTableChildComponentId(this.props.id, TableChildComponent.FILTER)}
+        key={getTableChildComponentId(this.props.id, TableChildComponent.FILTER)} />
+      : null;
+
+    const predicatesConnected: JSX.Element[] = actionBar && predicates
+      ? predicates.map((predicate: ITablePredicate, i: number) => {
+        const predicateId = `${getTableChildComponentId(this.props.id, TableChildComponent.PREDICATE)}-${predicate.attributeName}`;
+        const containerClasses = i ? ['ml1'] : [''];
+        return (
+          <DropdownSearchConnected
```

Comment:
I'm confused, why would we use a DropdownSearch if we turn off the search?

Reply:
because our `Dropdown` is pretty custom IIRC and the DropdownSearch currently has everything we need for this case except minor tweaks i've made to it...open to other ideas.

Reply:
but I agree it's not instinctive though.